src/jit/namedintrinsiclist.h

```diff
@@ -79,7 +79,15 @@ enum HWIntrinsicFlag : unsigned int
     HW_Flag_BaseTypeFromArg = 0x400,
 
     // Indicates compFloatingPointUsed does not need to be set.
-    HW_Flag_NoFloatingPointUsed = 0x800
+    HW_Flag_NoFloatingPointUsed = 0x800,
+
+    // Maybe IMM
+    // the intrinsic has either imm or Vector overloads
+    HW_Flag_MaybeIMM = 0x1000,
+
+    // NoJmpTable IMM
+    // the imm intrinsic does not need jumptable fallback when it gets non-const argument
+    HW_Flag_NoJmpTableIMM = 0x2000,
```

Comment:
I still don't understand why both of these are required. In what case would we require a jump table if there was a non-immediate form available, and conversely, in what case would there be a non-immediate form available but we still require a jump table? Are there cases where the values covered by an immediate are somehow broader or otherwise disjoint from the values covered by a non-immediate form? I note that the existing intrinsics you added always have neither or both of these flags set.

Reply:
> This is still not clear to me. All instances of one also include the other. Can you give an example that would not have both flags?

@CarolEidt The new-style intrinsic is based on function names rather than signatures,  and some intrinsics have "imm" and simple-SIMD overloads both. For example
```csharp
public static Vector256<sbyte> Shuffle(Vector256<sbyte> value, Vector256<sbyte> mask);
public static Vector256<int> Shuffle(Vector256<int> value, byte control) ;
```
so a category `HW_Category_IMM` is not enough to indicate a intrinsic requiring `imm8` or not (the above intrinsic should have flag `HW_Flag_MaybeIMM`). Consequently, we have to use the function `bool Compiler::isImmHWIntrinsic(NamedIntrinsic intrinsic, GenTree* lastOp)`.

In this kind of imm-intrinsic (`HW_Flag_MaybeIMM`), certain intrinsics are special, which have encoding that can carry the `imm8` operand in XMM registers, for example
```csharp
public static Vector256<ulong> ShiftRightLogical(Vector256<ulong> value, byte count);
public static Vector256<ulong> ShiftRightLogical(Vector256<ulong> value, Vector128<ulong> count);
                                                                          ^^^^^^
```
So, this intrinsic can have flag `HW_Flag_NoJmpTableIMM` to use XMM version as the non-constant fallback.

> I note that the existing intrinsics you added always have neither or both of these flags set.

Consequently, `Avx2.Shuffle` will need `HW_Flag_MaybeIMM` but no `HW_Flag_NoJmpTableIMM`.

Reply:
> So, this intrinsic can have flag HW_Flag_NoJmpTableIMM to use XMM version as the non-constant fallback (rather than a jump-table).

I believe that the ISA design (carry the imm8 operand in XMM registers) is intentional for this situation, and ARM also has similar instructions.

Reply:
OK, thanks for the explanation.